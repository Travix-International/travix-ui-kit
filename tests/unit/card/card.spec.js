import { shallow } from 'enzyme';
import React from 'react';
import Card from '../../../components/card/card';

describe('Card', () => {
  it('should return content inside Card container with default props', () => {
    const wrapper = shallow(
      <Card>
        Card Content
      </Card>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should return content inside Card container with checked state', () => {
    const wrapper = shallow(
      <Card checked>
        Card Content
      </Card>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should return content inside Card container with transparent state', () => {
    const wrapper = shallow(
      <Card transparent>
        Card Content
      </Card>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should return content inside Card container with checked and transparent state', () => {
    const wrapper = shallow(
      <Card checked transparent>
        Card Content
      </Card>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should return content inside Card container with hovering state', () => {
    const wrapper = shallow(
      <Card hovering>
        Card Content
      </Card>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should return content inside Card container with transparent and hovering state', () => {
    const wrapper = shallow(
      <Card hovering transparent>
        Card Content
      </Card>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should return content inside Card container with icon', () => {
    const wrapper = shallow(
      <Card checked showIcon>
        Card Content
      </Card>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should return content inside Card container with data attributes', () => {
    const wrapper = shallow(
      <Card checked dataAttrs={{ 'gtm-id': 'card1' }}>
        Card Content
      </Card>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
