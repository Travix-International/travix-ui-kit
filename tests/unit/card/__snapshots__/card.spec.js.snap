// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should return content inside Card container with checked and transparent state 1`] = `
<div
  className="ui-card ui-card_checked ui-card_transparent"
>
  Card Content
</div>
`;

exports[`Card should return content inside Card container with checked state 1`] = `
<div
  className="ui-card ui-card_checked"
>
  Card Content
</div>
`;

exports[`Card should return content inside Card container with default props 1`] = `
<div
  className="ui-card"
>
  Card Content
</div>
`;

exports[`Card should return content inside Card container with hovering state 1`] = `
<div
  className="ui-card ui-card_hovering"
>
  Card Content
</div>
`;

exports[`Card should return content inside Card container with icon 1`] = `
<div
  className="ui-card ui-card_checked ui-card_show-icon"
>
  Card Content
</div>
`;

exports[`Card should return content inside Card container with transparent and hovering state 1`] = `
<div
  className="ui-card ui-card_transparent ui-card_hovering"
>
  Card Content
</div>
`;

exports[`Card should return content inside Card container with transparent state 1`] = `
<div
  className="ui-card ui-card_transparent"
>
  Card Content
</div>
`;
